hermetic_build/common/cli/changed_library_generation.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+import click as click
+
+from common.model.generation_config import from_yaml
+from common.utils.generation_config_comparator import compare_config
+
+
+@click.group(invoke_without_command=False)
```

Comment:
Since this file is not supposed to be called directly, does it still have to be a cli? Maybe cli is still easier, but if not, can we call it as a Python function directly?

Reply:
We need to use this function in shell script, so I add a CLI to simplify the call.

The result is passed to `entry_point.py`.

Reply:
This function is used in shell script so I add a CLI to simplify the call.